redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,69 @@
+package redisxtest
+
+import (
+	"context"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// MockRedisCluster wraps a local version of redis
+type MockRedisCluster struct {
+	redisCluster *miniredis.Miniredis
+}
+
+func NewMockRedisCluster() (mockRedisCluster MockRedisCluster, teardown func(), err error) {
+	redisCluster, err := miniredis.Run()
+	if err != nil {
+		return MockRedisCluster{}, nil, err
+	}
+
+	mockRedisCluster = MockRedisCluster{
+		redisCluster: redisCluster,
+	}
+
+	teardown = func() {
+		mockRedisCluster.Close()
+	}
+
+	return mockRedisCluster, teardown, nil
+}
+
+// Addr returns address of mock redis cluster e.g. '127.0.0.1:12345'.
+func (mrc *MockRedisCluster) Addr() string {
+	return mrc.redisCluster.Addr()
+}
+
+// Close shuts down the MockRedisCluster
+func (mrc *MockRedisCluster) Close() {
```

Comment:
nit: make this function to return an error, so it implements [`io.Closer`](https://pkg.go.dev/io#Closer) interface

since `miniredis.Miniredis.Close` does not return an error, you can just always `return nil` inside this function.